src/terminatesteadystate.jl

```diff
@@ -15,8 +15,7 @@ function allDerivPass(integrator, abstol, reltol)
         end
     end
 
-    any(abs(d) > abstol && abs(d) > reltol*abs(u) for (d,abstol, reltol, u) =
-           zip(testval, Iterators.cycle(abstol), Iterators.cycle(reltol), integrator.u)) && (return false)
+    any((abs.(testval) .> abstol) .& (abs.(testval) .> reltol .* abs.(integrator.u))) && (return false)
```

Comment:
Maybe we can at least do, `if typeof(integrator.u) <: Array` and have a fast path there?

Reply:
That won't work for an `ArrayPartition` though 🤔

Reply:
You want to broadcast ArrayPartition though?

Reply:
Yes if it is an `ArrayPartition` of GPU arrays. I must use `ArrayPartition` because the GPU arrays have different shapes.

Reply:
That isn't the point. See the change I made. Everything broadcasts except Array. Array just loops optimally, everything else broadcasts, ArrayPartition of GPUs and all of that will work just fine.

Reply:
Probably more efficient for ArrayPartition because its linear indexing is slow. But there are cases like `LArray` which 🤷 we can handle that if someone truly cares about a 10% performance boost to LArray. For now I'm happy.